zone/spells.cpp

```diff
@@ -229,6 +229,16 @@ bool Mob::CastSpell(uint16 spell_id, uint16 target_id, CastingSlot slot,
 		return false;
 	}
 
+	if (IsEffectInSpell(spell_id, SE_Charm) && !PassCharmTargetRestriction(entity_list.GetMobID(target_id))) {
+		if (IsClient()) {
+			CastToClient()->SendSpellBarEnable(spell_id);
```

Comment:
We don't need to cast to client here. We should also make sure the charm is a casted spell before calling SendSpellBarEnable.